src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java

```diff
@@ -237,72 +257,159 @@ public void addAssessment(long exerciseId, long submissionId, List<Feedback> mod
+    /**
+     * Update the (existing) automatic result for the given submission with automatic feedback generated by Compass. If there is no existing result, a new one is created first. The
+     * updated result gets added to the hash map containing all automatic results. Note, that Compass tries to automatically assess every model as much as possible, but does not
+     * submit any automatic assessment to the student. A user has to review every(!) automatic assessment before completing and submitting the assessment manually, even if Compass
+     * could assess 100% of the model automatically.
+     *
+     * @param submissionId the id of the modeling submission for which an automatic result should be generated/updated
+     * @param exerciseId   the id of the corresponding exercise
+     */
+    private void assessAutomatically(long submissionId, long exerciseId) {
+        // Workaround for ignoring automatic assessments of unsupported modeling exercise types TODO remove this after adapting compass
+        if (!isSupported(exerciseId)) {
+            return;
+        }
+
+        CalculationEngine engine = compassCalculationEngines.get(exerciseId);
+        ModelingSubmission modelingSubmission = findModelingSubmissionById(submissionId);
+
+        if (engine == null || modelingSubmission == null) {
+            log.error("No calculation engine or submission - submission with ID {} could not be assessed automatically", submissionId);
+            return;
+        }
+
+        Result result = provideResultForSubmission(modelingSubmission);
+        generateAutomaticResult(submissionId, result, engine);
+    }
+
+    /**
+     * Update the (existing) automatic result for each of the given submissions with automatic feedback generated by Compass. If there is no existing result for a submission, a new
+     * one is created first. The updated results get added to the hash map containing all automatic results. Note, that Compass tries to automatically assess every model as much as
      * possible, but does not submit any automatic assessment to the student. A user has to review every(!) automatic assessment before completing and submitting the assessment
      * manually, even if Compass could assess 100% of the model automatically.
      *
-     * @param modelId    the id of the model/submission that should be updated with an automatic assessment
-     * @param exerciseId the id of the corresponding exercise
+     * @param submissionIds a collection of modeling submission ids for which the automatic results should be generated/updated
+     * @param exerciseId    the id of the corresponding exercise
      */
-    private void assessAutomatically(long modelId, long exerciseId) {
+    private void assessAllAutomatically(Collection<Long> submissionIds, long exerciseId) {
+        // Workaround for ignoring automatic assessments of unsupported modeling exercise types TODO remove this after adapting compass
+        if (!isSupported(exerciseId)) {
```

Comment:
see above

Reply:
I added the `isSupported` check to all relevant public methods of the `CompassService` and removed it from the private methods.